Lib/asyncio/coroutines.py

```diff
@@ -19,8 +19,11 @@ def _is_debug_mode():
 
 def iscoroutinefunction(func):
     """Return True if func is a decorated coroutine function."""
-    return (inspect.iscoroutinefunction(func) or
-            getattr(func, '_is_coroutine', None) is _is_coroutine)
+    return (
+        inspect.iscoroutinefunction(func)
+        or getattr(func, '_is_coroutine', None) is _is_coroutine
```

Comment:
this needs to both check the attribute and unwrap the partial so it can support code that worked around this issue by doing:

```
        def sync_fn():
            pass

        partial_sync_fn = functools.partial(sync_fn)
        partial_sync_fn._is_coroutine = asyncio.coroutines._is_coroutine
```